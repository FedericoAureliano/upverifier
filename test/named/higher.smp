data U := A | B | C

function x : Seq[U] := unit(A) ^ unit(B) 
function y : Seq[U] := unit(A) ^ unit(A) ^ unit(B)

assume(|y| > |x|)
check
assume(contains(x, y))
check
assume(suffix(x, y))
check
assume(prefix(x, y))
check
