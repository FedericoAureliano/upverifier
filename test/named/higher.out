CHECK: true
CHECK: true
CHECK: true
CHECK: false
